CommandLine/libEGM/tmx.h

```diff
@@ -227,7 +256,8 @@ class TMXMapLoader {
    * @return Returns true if parsing succeeds else false
    */
   bool LoadCsvLayerData(const std::string &dataStr, buffers::TreeNode *resNode, const int &tileWidth,
-                        const int &tileHeight, const int &layerWidth, const int &layerHeight);
+                        const int &tileHeight, const int &layerWidth, const int &layerHeight, const int &chunkXIdx = 0,
+                        const int &chunkYIdx = 0);
```

Comment:
Is size of `const int &` is 8 bytes? I just did that going by the general idea of using `const` wherever possible. And reference is to make copy cheap.

Reply:
Well, you can't make copying an `int` any cheaper. It's 32-bit, so 4 bytes, whereas a reference, which is really the same as a pointer, is nowadays usually 64-bit, so 8 bytes. In addition to passing around more bytes to do the function call, you're adding a level of indirection. This means that in the function, it will need to use the reference to get the actual value.

Using references makes sense for huge structures, or even to avoid heap allocations (as would be the case when passing `std::string` by value), but it makes no sense for small types that are fast to copy.

Reply:
Nice, I never learned this before. Will fix that.